js/packages/cli/src/helpers/upload/ipfs.ts

```diff
@@ -0,0 +1,58 @@
+import log from "loglevel";
+import fetch from 'node-fetch';
+import { create, globSource } from 'ipfs-http-client';
+
+export interface ipfsCreds {
+    projectId: string,
+    secretKey: string
+}
+
+function sleep(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+export async function ipfsUpload(ipfsCredentials: ipfsCreds, image: string, manifestBuffer: Buffer) {
+  const tokenIfps = `${ipfsCredentials.projectId}:${ipfsCredentials.secretKey}`;
+  // @ts-ignore
+  const ipfs = create("https://ipfs.infura.io:5001");
+
+  const uploadToIpfs = async (source) => {
+    const { cid } = await ipfs.add(source).catch();
+    return cid;
+  };
+
+  const mediaHash = await uploadToIpfs(globSource(image, { recursive: true }));
+  log.debug('mediaHash:', mediaHash);
+  const mediaUrl = `https://ipfs.io/ipfs/${mediaHash}`;
+  log.debug('mediaUrl:', mediaUrl);
+  const authIFPS = Buffer.from(tokenIfps).toString("base64");
+  await fetch(`https://ipfs.infura.io:5001/api/v0/pin/add?arg=${mediaHash}`, {
```

Comment:
Just as a side note to this, I think it is possible to use this `fetch` to upload and pin in the same call. It would save a network request and I think as it is authenticated, it could also be possible that we can also get rid of the `sleep`s, I am unsure if IPFS rate-limits authenticated users.

Reply:
I couldn't figure out how to best teach the`ipfs-http-client` how to authenticate and/or pin

Reply:
Yes they were purely there to prevent 429s

Reply:
possible i might do this tomorrow, will report

Reply:
Hello sirs, how can I use this for my NFT project. I want to upload to IPFS

Reply:
hey @KingFavourCreates - we'll merge soon! thanks for your interest ☺️